src/server/game/Maps/Map.cpp

```diff
@@ -2899,6 +2903,103 @@ ZLiquidStatus Map::getLiquidStatus(PhaseShift const& phaseShift, float x, float
     return result;
 }
 
+void Map::GetFullTerrainStatusForPosition(float x, float y, float z, PositionFullTerrainStatus& data, uint8 reqLiquidType) const
+{
+    VMAP::IVMapManager* vmgr = VMAP::VMapFactory::createOrGetVMapManager();
+    VMAP::AreaAndLiquidData vmapData;
+    vmgr->getAreaAndLiquidData(GetId(), x, y, z, reqLiquidType, vmapData);
```

Comment:
you need to add `uint32 terrainMapId = PhasingHandler::GetTerrainMapId(phaseShift, this, x, y);` and replace `GetId()` with it

also drop `const` from function signature (been done for all Map functions that use GetGrid)